sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -2200,9 +2200,9 @@ ProgramManager::link(const std::vector<device_image_plain> &DeviceImages,
 
   DeviceImageImplPtr ExecutableImpl =
       std::make_shared<detail::device_image_impl>(
-          /*BinImage=*/nullptr, Context, Devs, bundle_state::executable,
-          std::move(KernelIDs), LinkedProg, std::move(NewSpecConstMap),
-          std::move(NewSpecConstBlob));
+          getSyclObjImpl(DeviceImages[0])->get_bin_image_ref(), Context, Devs,
```

Comment:
Can we also change the signature of ProgramManager::link to accept a single device image to make this assumption explicit? If I understand correctly, it's the only current use case anyway.

Reply:
I'm confused a little as it makes not small changes to this function, that may be required in [future](https://github.com/intel/llvm/blob/sycl/sycl/source/detail/kernel_bundle_impl.hpp#L197) - but okay [79c19ac](https://github.com/intel/llvm/pull/9085/commits/79c19ac373847e9919d4af49482e6891fa5cf61c).